src/material-experimental/theming/_theming.scss

```diff
@@ -0,0 +1,142 @@
+@use 'sass:list';
+@use 'sass:map';
+@use 'sass:meta';
+@use 'sass:string';
+@use '@angular/material' as mat;
+
+// Whether to throw an error when a required dep is not configured. If false, the dep will be
+// automatically configured instead.
+$_error-on-missing-dep: false;
+
+// Applies the theme for the given component configuration.
+@mixin _apply-theme($tokens, $component) {
+  $id: map.get($component, id);
+  $tokens: map.deep-merge($tokens, map.get($component, customizations));
+
+  // NOTE: for now we use a hardcoded if-chain, but in the future when first-class mixins are
+  // supported, the configuration data will contain a reference to its own theme mixin.
+  @if $id == 'mat.card' {
+    @include mat.card-theme($tokens);
+  } @else if $id == 'mat.checkbox' {
+    @include mat.checkbox-theme($tokens);
+  } @else {
+    @error 'Unrecognized component theme: #{id}';
+  }
+}
+
+// Gets the transitive dependency configurations for the given list of component configurations.
+@function _get-transitive-deps($components, $configured: ()) {
+  // Mark the given components as configured.
+  @each $component in $components {
```

Comment:
Am I interpreting this function correctly that order doesn't matter? The variables for everything can be emitted in any order, so we only care about the presence of deps?

(if so it would be good to add a comment to that effect)

Reply:
Yeah basically. The one thing I can think of where we might care about the order is if multiple configurations are trying to customize the same token, which one winds up taking precedence. My thinking for the most part id that we don't want multiple functions trying to set the same token, but I think it would be worth trying to think of some scenarios where it might come up and whether there's a particular strategy for which gets precedence that makes more sense.

I'll add a comment and a TODO to come back and think about this more